### ISSUE TITLE
Clean up test Clock and move it to a public location

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1757]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1757|
|Type|enhancement|
|Created|2006-05-29 16:42:25Z|

Description

<details><summary>Searchable metadata</summary>

```
trac-id__1757 1757
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1148920945000000 1148920945000000
changetime__1149373166000000 1149373166000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @glyph|
|-|-|

Ready for review in test-clock-1757

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @exarkun|
|-|-|

The docstring for 'seconds' is inadequate; am I supposed to subclass and override this?  Does it have any actual relationship to time.time()?

```callLater``` has no docstring, and that seems pretty important.

Does this thing actually want to implement ```IReactorTime```?

```adjust``` does not document the type of its parameter.  Also it doesn't really distinguish itself from ```advance``` very well.

Is there really a reason for ```_runUntilCurrent``` to be a separate method?  It's only called in one place, and it's private.

If you want to merge as-is for compatibility reasons, go ahead: but please create some tickets for documenting this.  We should probably have a document that explains the purpose and method for faking out both the reactor and the clock (and the fake reactor in Vertex's tests should move when Juice does, so perhaps that's a dependency too...)

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set status to `closed`|
|-|-|

Fixed in r17029.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `reopened`|
|-|-|

Tests failed on Python 2.3, reverted at r17030

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [17032]) Merge test-clock-1757

Authors: exarkun, radix
Reviewer: glyph, radix
Fixes [#1757](https://github.com/twisted/twisted/issues/1757)


This adds a twisted.internet.task.Clock class which
simulates time as an IReactor.callLater replacement.
It is useful for unit tests.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
